### PR TITLE
add prepare command to package tooling

### LINKED ISSUE #2681: add prepare command to package tooling
**Describe the feature request**
We need a  `prepare` command that can be used to do any pre-reqs needed to use package tooling. We can start with downloading carvel binaries needed to build packages for now and make any additions if needed later. This command will be useful when we pull out the package tooling go module into a new git repository

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by yharish991**
Signed-off-by: Harish Yayi <yharish991@gmail.com>

### What this PR does / why we need it

This PR adds a prepare command that can be used to do any pre-reqs needed to use package tooling. Right now it downloads carvel binaries needed to build packages. This command will be useful when we pull out package tooling go module into a new git repository

### Which issue(s) this PR fixes

Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/2681

### Describe testing done for PR
Run the below command and verify that the new carvel binaries are downloaded

```
cd hack/packages/package-tools && go run main.go prepare --clean
```

### Release note

```release-note
Added `prepare` command to package tooling
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: hack/packages/package-tools/cmd/prepare.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+				return fmt.Errorf("couldn't delete carvel binaries: %w", err)
+			}
+		}
+	}
+
+	if err := utils.CreateDir(toolsBinDir); err != nil {
+		return err
+	}
+
+	for _, tool := range c.Tools {
```
Comment (rajathagasthya) [inline hack/packages/package-tools/cmd/prepare.go:75]: optional: we could use concurrency to speed this up, but also okay if you want do it later.

**Review by rajathagasthya — COMMENTED** [review]
Minor comments, but overall LGTM.

File: hack/packages/package-tools/cmd/prepare.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+//go:embed config/carvelTools-config.yaml
+var carvelTools []byte
+
+var clean bool
+
+func init() {
+	rootCmd.AddCommand(prepareCmd)
+	prepareCmd.Flags().BoolVar(&clean, "clean", false, "Deletes the existing Carvel tool binaries")
```
[the remark this answers is not in the document]
Comment (yharish991) [reply hack/packages/package-tools/cmd/prepare.go:38]: clean flag simply deletes the binaries, I don't think it's worth making it a separate command.

File: hack/packages/package-tools/cmd/prepare.go (hunk 1 above)
Comment (yharish991) [reply hack/packages/package-tools/cmd/prepare.go:76]: sure, we can use concurrency here, but right now I think it's not worth it as we are downloading just 4 binaries

**Review by rajathagasthya — APPROVED** [review]

**Pull request merged**